### PR TITLE
Check metric type before search

### PR DESCRIPTION
**Pull request by bigsheeper**
issue: https://github.com/milvus-io/milvus/issues/24533

/kind enhancement

### CODE REVIEW

**Comment by bigsheeper** [comment]
test:

image

File: internal/querynodev2/services.go [hunk 1]
```diff
@@ -757,13 +757,26 @@ func (node *QueryNode) Search(ctx context.Context, req *querypb.SearchRequest) (
 			ErrorCode: commonpb.ErrorCode_Success,
 		},
 	}
+	collection := node.manager.Collection.Get(req.GetReq().GetCollectionID())
+	if collection == nil {
+		failRet.Status.ErrorCode = commonpb.ErrorCode_UnexpectedError
+		failRet.Status.Reason = merr.WrapErrCollectionNotFound(req.GetReq().GetCollectionID()).Error()
```
Comment (yah01) [inline internal/querynodev2/services.go:763]: failRet.Status = merr.Status(merr.WrapErrXxx(...))

File: internal/querynodev2/services.go [hunk 2]
```diff
@@ -757,13 +757,26 @@ func (node *QueryNode) Search(ctx context.Context, req *querypb.SearchRequest) (
...
+		return failRet, nil
+	}
+
+	// Check if the metric type specified in search params matches the metric type in the index info.
+	if !req.GetFromShardLeader() && req.GetReq().GetMetricType() != "" {
+		if req.GetReq().GetMetricType() != collection.GetMetricType() {
+			failRet.Status.ErrorCode = commonpb.ErrorCode_UnexpectedError
+			failRet.Status.Reason = merr.WrapErrParameterInvalid(collection.GetMetricType(), req.GetReq().GetMetricType(),
```
Comment (yah01) [inline internal/querynodev2/services.go:771]: Same as line 763

**Review by yah01 — CHANGES_REQUESTED** [review]

**Comment by bigsheeper** [comment]
also related: https://github.com/milvus-io/milvus/issues/24555

**Pull request merged**